dart/common/Uri.h

```diff
@@ -97,16 +100,34 @@ class Uri final
   /// Fragment, e.g. the part of the URI after the #
   UriComponent mFragment;
 
+  /// Constructor
+  Uri() = default;
+
+  /// Constructor that takes a URI string. Internally, this is equivalent to
+  /// calling fromString(_input) after default constructor.
+  Uri(const std::string& _input);
```

Comment:
As @mxgrey suggested:

> Have an implicit constructor for the Uri class that takes in a string an converts it to a Uri (equivalent to the existing fromString(~) function). This should allow backwards compatibility.

I intended it expecting following code would work:
```cpp
auto world = SkelParser::readWorld("/my/world.skel");  // error
```
but didn't, and following worked:
```cpp 
auto world = SkelParser::readWorld(std::string("/my/world.skel"));  // ok
```

How can I make the first code to work?

Also, I wonder if this would allow backward compatibility even the first code works because `fromString(~)` only correctly handles URI strings but not raw paths (especially Windows style paths). Instead, I left old functions that take filename as deprecated and made them to call the corresponding new functions that take `Uri`.

Here is some cases that should work I think:
```cpp
auto world = SkelParser::readWorld(Uri::createFromPath("/my/world.skel"));  // ok
auto world = SkelParser::readWorld(Uri::createFromString("file:///my/world.skel"));  // ok
auto world = SkelParser::readWorld(Uri::createFromString("file:///C:/my/world.skel"));  // ok
auto world = SkelParser::readWorld(Uri("file:///my/world.skel"));  // ok
auto world = SkelParser::readWorld("file:///my/world.skel");  // possible?
```